cEP-0016.md

```diff
@@ -0,0 +1,234 @@
+body_close_issue_full_url = True
+body_close_issue_on_last_line = True
+body_enforce_issue_reference = False
+```
+
+```
+[commit]
+bears = `GitCommitBear`
+```
+
+## Proposed Change
+
+This project would involve making three meta bears that would identify
+commits that skip CI buids, `git merge` and `git revert` commits.
+
+Here is the detailed implementation stepwise:
+
+1. We start by identifying different types of special commits namely Travis
+   merge commits, `git revert` , `git merge` and commits that disable CI builds.
+2. Each type of commit is then handled as a separate case.
+3. For Travis commits:
+   Travis commits are unseen merge commits that are sent to CI when GitHub merge
+   commits are added to 3+ neatly stacked PRs. These merge commits are different
+   as they are not made using `git merge`. Instead they are made by Github
+   automatically when a PR is commited by more than two people. `GitCommitBear`
+   at present does not identify such commits and thus fails. This failure is
+   then reflected in Travis builds causing it to fail as well.
+   - As a solution to this problem, settings will be added in `GitCommitBear`
+     that would identify Travis merge commits and ignore them. Therefore, the
+     `GitCommitBear` will run without failing and so will the Travis build.
+4. For commits that disable CI builds:
+   coala uses Travis CI, Circle CI, AppVeyor, GitLab CI and Semaphore CI for
+   continuous integration. Most of these support `[ci skip]` and `[skip ci]` for
+   skipping a build for a particular commit. Commits that have `[ci skip]` or
+   `[skip ci]` anywhere in the commit messages are ignored by them. With this
+   project, the `GitCommitBear` would support skipping builds for CI engines
+   that use these special sequences for skipping jobs. For others that do not
+   support such commands, the bear would give an appropriate message.
+   - Commit contents are checked to find `[ci skip]`.
+   - For such commits, it is checked that the commit body explains a valid
+     reason why this change should not include CI builds.
+   - Files modified in the commit should include files that are not covered by
+     `.coafile` or tests. If this is not the case, an error is raised to exclude
+     `[ci skip]` from the commit message.
+5. For merge commits:
+   - Merge commits are identified by checking the parents of the commit.
+   - In case the bear finds that the commit tries to merge master onto the pull
+     request, message is displayed to perform a `git rebase` instead.
+6. For revert commits:
+   - Revert commits are identified based on the format of the commit.
+   - For revert commits, the patches of both commits are compared to ensure that
+     they are in fact reverse of each other. Minor differences are allowed.
+   - In case of total mismatch, a note in the commit message highlighting that
+     it is not a clean revert is expected.
+   - A message suggesting improvements in the commit message is displayed.
```

Comment:
you introduce this section by saying you're creating *three* meta bears. Even though I don't exactly know what a meta bear is, I'm supposing it's a variation of the `GitCommitBear` that handles a specific use case. But you talk about four use cases.

So is there a meta bear that will handle two different use cases? if so, which ones?